janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/util/RestClientAuthenticator.java

```diff
@@ -0,0 +1,28 @@
+package org.janusgraph.diskstorage.es.rest.util;
+
+import java.io.IOException;
+
+import org.elasticsearch.client.RestClientBuilder.HttpClientConfigCallback;
+import org.elasticsearch.client.RestClientBuilder.RequestConfigCallback;
+
+/**
+ * <p>
+ * A class instance implementing this interface may customize some aspects of
+ * JanusGraph ES REST client to implement a custom authentication method.
+ * </p>
+ * <p>
+ * A class implementing instance must provide a constructor that accepts a String[]. This
+ * array will contain the values provided as constructor parameters in JanusGraph graph index
+ * backend configuration. If the list of parameters is empty, a zero-size array is passed to the
```

Comment:
Replace "JanusGraph graph index backend configuration" with `"{@link ElasticSearchIndex#ES_HTTP_AUTHENTICATOR_ARGS}"`? Would need to import ElasticSearchIndex if you do that though.

Reply:
not with the fully qualified name - only the javadoc will need the class I refer to :)

Reply:
updated